demo/src/main/java/com/eis/geoCalendar/demo/Localization/LocationManager.java

```diff
@@ -0,0 +1,140 @@
+ *
+ * TODO: this is here temporarely just for the demo
+ */
+public class LocationManager {
+    private static final String[] permissions = {
+            Manifest.permission.ACCESS_COARSE_LOCATION,
+            Manifest.permission.ACCESS_FINE_LOCATION,
+            Manifest.permission.READ_PHONE_STATE,
+    };
+
+
+    private final String LocationManagerTag = "LocationManagerTag";
+    private final String MAPS_START_URL = "https://www.google.com/maps/search/?api=1&query=";
+    //NOTE: concat latitude,longitude
+
+    private Context currentContext;
+    private LocationRequest locationRequest;
+    private PendingIntent locationIntent;
+    private Location mLastLocation;
+
+    private FusedLocationProviderClient mFusedLocationClient;
+
+    public LocationManager(Context applicationContext){
+        currentContext = applicationContext;
+    }
+
+    /***
+     * Method that gets the last Location available of the device, and executes the imposed command
+     * calling command.execute(foundLocation)
+     *
+     * @param command object of a class that implements interface Command
+     */
+    public void getLastLocation(final Command<Location> command)
+    {
+        mFusedLocationClient = LocationServices.getFusedLocationProviderClient(currentContext);
+        mFusedLocationClient.flushLocations();
+        locationRequest = LocationRequest.create();
+        locationRequest.setPriority(PRIORITY_HIGH_ACCURACY);
+
+        mFusedLocationClient.getLocationAvailability().addOnSuccessListener(new OnSuccessListener<LocationAvailability>() {
+            @Override
+            public void onSuccess(LocationAvailability locationAvailability) {
+                Log.d(LocationManagerTag, "onSuccess: locationAvailability.isLocationAvailable " + locationAvailability.isLocationAvailable());
+
+                mFusedLocationClient.requestLocationUpdates(locationRequest, locationIntent)
+                        .addOnCompleteListener(new OnCompleteListener() {
+                            @Override
+                            public void onComplete(@NonNull Task task) {
+                                Log.d(LocationManagerTag, "Update Result: " + task.getResult());
+                            }
+                        });
+
+                Log.d(LocationManagerTag, "Requested updated location: ");
+
+                mFusedLocationClient.getLastLocation().addOnCompleteListener(new OnCompleteListener<Location>() {
+                    @Override
+                    public void onComplete(@NonNull Task<Location> task) {
+                        Log.d(LocationManagerTag, "Completed lastLocation");
+                        Log.d(LocationManagerTag, "Task<Location> successful " +  task.isSuccessful());
+
+                        if (task.isSuccessful() && task.getResult() != null) {
+                            mLastLocation = task.getResult();
+                            Log.d(LocationManagerTag, "Victory!" +mLastLocation.toString());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['+' is not followed by whitespace.](https://app.codacy.com/manual/CremaLuca/geoCalendar/pullRequest?prid=4824932)